lib/puma/binder.rb

```diff
@@ -229,8 +224,14 @@ def parse(binds, logger, log_msg = 'Listening')
             io = inherit_ssl_listener sock, ctx
             logger.log "* Activated #{str}"
           else
+            ios_len = @ios.length
             io = add_ssl_listener uri.host, uri.port, ctx
-            logger.log "* Listening on #{str}"
+
+            ssl_qry = str[/\?.+\z/]
+            @ios[ios_len..-1].each do |i|
+              addr = loc_addr_str i
+              logger.log "* #{log_msg} on ssl://#{addr}#{ssl_qry}"
```

Comment:
Can't we use the URI object to get the query instead of the regexp?

```suggestion
              logger.log "* #{log_msg} on ssl://#{addr}?#{uri.query}"
```

Reply:
Good idea.  Changed...